napari/layers/utils/layer_utils.py

```diff
@@ -907,6 +909,11 @@ def _validate_features(
     """
     if isinstance(features, pd.DataFrame):
         features = features.reset_index(drop=True)
+    elif isinstance(features, dict):
+        # One needs to reset pd.Series object's indices as well
+        for key, value in features.copy().items():
+            if isinstance(value, pd.Series):
+                features[key] = value.reset_index(drop=True)
```

Comment:
```suggestion
        features = {key: np.array(value, copy=False)}
```

This should handle all array likes and things that implement `__array__`, which includes `Series`. I think `copy=False` is fine here as we don't make guarantees about copies and this should avoid an unnecessary copy when a `np.ndarray` was passed.

Reply:
Might also be good to include an implementation/explanatory comment like you did with a link to this PR.